### DISCUSSION TITLE
[aws-resources] EKS resource and pod information

### DISCUSSION THREAD

**Discussion by BT-22** [open-telemetry/opentelemetry-java-contrib#577] [Q&A]
Hello, according to https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-aws  

```
eks – Information about an Amazon EKS cluster.

- pod – The hostname of your EKS pod.
- cluster_name – The EKS cluster name.
- container_id – The full container ID of your container.
```
However, in X-Ray trace JSON data, under `aws` it does not seem to have those 3 pieces of information under `eks`

Below https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EksResource.java#L67 
It appears only `cloud.provider` and `cloud.platform` are collected always.
`cluster name` and `container Id` are also collected but conditional.
I do not see `pod name`.

I'm wondering if under `aws` then under `eks` is where I should expect to find pod name and container ID?
Or I should expect finding them elsewhere?

Also, is there some configuration setting we need in order to see pod information in the X-Ray trace?
I only see `cloud.provider` and `cloud.platform` in our traces...

Thanks!

**Comment by trask** [comment]
hi @BT-22! we aren't generally capable to answer/support questions related to specific monitoring backends (in this case X-Ray).

copying @willarmiros who may be able to help redirect/answer in this particular case.

**Comment by willarmiros** [comment]
Hi @BT-22 - please see our translation docs to understand which OTel resource attributes are converted to those `aws.eks` attributes on the X-Ray Console: https://aws-otel.github.io/docs/getting-started/x-ray#otel-span-eks-metadata-translation

Generally speaking, the EKS resource detector should record *at least* these 3 attributes in each language it's implemented in, and if it doesn't that should be considered a bug and I would encourage you to open an issue on the repo where you see that missing behavior.

**Reply by BT-22** [reply]
Thank you @willarmiros 
Acknowledge your reply. I'll investigate further and if I'm 100% sure, then I'll open an issue on the repo.

I recently found someone removed some attribute in our X-Ray collector and caused loss of stack trace.
So, I'll need to test again with "vanilla" X-Ray collector to avoid issue due to other removed attributes.

Will give an update here of my findings.
